hadoop-common-project/hadoop-auth/src/test/java/org/apache/hadoop/security/authentication/util/TestFileSignerSecretProvider.java

```diff
@@ -48,4 +51,22 @@ public void testGetSecrets() throws Exception {
     Assert.assertEquals(1, allSecrets.length);
     Assert.assertArrayEquals(secretValue.getBytes(), allSecrets[0]);
   }
+
+  @Test
+  public void testEmptySecretFileThrows() throws Exception {
+    File secretFile = File.createTempFile("test_empty_secret", ".txt");
+    assertTrue(secretFile.exists());
+
+    FileSignerSecretProvider secretProvider
+            = new FileSignerSecretProvider();
+    Properties secretProviderProps = new Properties();
+    secretProviderProps.setProperty(
+            AuthenticationFilter.SIGNATURE_SECRET_FILE,
+            secretFile.getAbsolutePath());
+
+    Exception exception = assertThrows(RuntimeException.class, () ->
```

Comment:
As this patch could be a potential candidate for a branch 2.10 backport (if anyone needs it of course), using lambda expressions is generally discouraged. I personally find it useful here and there, but it is really divising among the community. My advice would be to use it whenever you have a huge gain by its usage, in order to reduce friction between community members.

Reply:
Good to know, replaced it with a ThrowingRunnable.